boutdata/tests/make_test_data.py

```diff
@@ -0,0 +1,390 @@
+        var = outputfile.createVariable(name, float, ("t", "x", "y", "z"))
+
+        data = rng.random((nt, localnx, localny, localnz))
+        var[:] = data
+        for key, value in expected_attributes[name].items():
+            var.setncattr(key, value)
+
+        result[name] = data[:, xslice, yslice, zslice]
+
+    create3D_t("field3d_t_1")
+    create3D_t("field3d_t_2")
+
+    def create3D(name):
+        var = outputfile.createVariable(name, float, ("x", "y", "z"))
+
+        data = rng.random((localnx, localny, localnz))
+        var[:] = data
+        for key, value in expected_attributes[name].items():
+            var.setncattr(key, value)
+
+        result[name] = data[xslice, yslice, zslice]
+
+    create3D("field3d_1")
+    create3D("field3d_2")
+
+    # Field2D
+    def create2D_t(name):
+        var = outputfile.createVariable(name, float, ("t", "x", "y"))
+
+        data = rng.random((nt, localnx, localny))
+        var[:] = data
+        for key, value in expected_attributes[name].items():
+            var.setncattr(key, value)
+
+        result[name] = data[:, xslice, yslice]
+
+    create2D_t("field2d_t_1")
+    create2D_t("field2d_t_2")
+
+    def create2D(name):
+        var = outputfile.createVariable(name, float, ("x", "y"))
+
+        data = rng.random((localnx, localny))
+        var[:] = data
+        for key, value in expected_attributes[name].items():
+            var.setncattr(key, value)
+
+        result[name] = data[xslice, yslice]
+
+    create2D("field2d_1")
+    create2D("field2d_2")
+
+    # FieldPerp
+    def createPerp_t(name):
+        var = outputfile.createVariable(name, float, ("t", "x", "z"))
+
+        data = rng.random((nt, localnx, localnz))
+        var[:] = data
+        for key, value in expected_attributes[name].items():
+            var.setncattr(key, value)
+        var.setncattr("yindex_global", fieldperp_global_yind)
+
+        result[name] = data[:, xslice, zslice]
+
+    createPerp_t("fieldperp_t_1")
+    createPerp_t("fieldperp_t_2")
+
+    def createPerp(name):
+        var = outputfile.createVariable(name, float, ("x", "z"))
+
+        data = rng.random((localnx, localnz))
+        var[:] = data
+        for key, value in expected_attributes[name].items():
+            var.setncattr(key, value)
+        var.setncattr("yindex_global", fieldperp_global_yind)
+
+        result[name] = data[xslice, zslice]
+
+    createPerp("fieldperp_1")
+    createPerp("fieldperp_2")
+
+    # Time-dependent array
+    def createScalar_t(name):
+        var = outputfile.createVariable(name, float, ("t",))
+
+        data = rng.random(nt)
+        var[:] = data
+
+        result[name] = data
+
+    createScalar_t("t_array")
+    createScalar_t("scalar_t_1")
+    createScalar_t("scalar_t_2")
+
+    # Scalar
+    def createScalar(name, value):
+        var = outputfile.createVariable(name, type(value))
+
+        var[...] = value
+
+        result[name] = value
+
+    createScalar("BOUT_VERSION", 4.31)
```

Comment:
I guess the version number needs to be updated manually if needed?

Reply:
Even better might be to parameterise it. At the moment though, I think the only reason it needs to be there is that there's a branch to support BOUT++ versions 3 and earlier (which removes the last point in the z-direction), so we need a version number >=4. If we add more version-specific behaviour in future it might be good to actually test that.

Reply:
That makes sense 🙂